examples/camera-controller/access-control/AccessControl.cpp

```diff
@@ -0,0 +1,95 @@
+/**
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "AccessControl.h"
+
+#include <access/AccessControl.h>
+#include <access/Privilege.h>
+#include <access/ProviderDeviceTypeResolver.h>
+#include <access/RequestPath.h>
+#include <access/SubjectDescriptor.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/InteractionModelEngine.h>
+#include <lib/core/CHIPError.h>
+#include <lib/core/Global.h>
+
+using namespace chip;
+using namespace chip::Access;
+using namespace chip::app::Clusters;
+
+namespace {
+
+class DeviceTypeResolver : public chip::Access::DynamicProviderDeviceTypeResolver
+{
+public:
+    DeviceTypeResolver() :
+        chip::Access::DynamicProviderDeviceTypeResolver(
+            [] { return chip::app::InteractionModelEngine::GetInstance()->GetDataModelProvider(); })
+    {}
+};
+
+class AccessControlDelegate : public Access::AccessControl::Delegate
+{
+    CHIP_ERROR Check(const SubjectDescriptor & subjectDescriptor, const RequestPath & requestPath,
+                     Privilege requestPrivilege) override
+    {
+        // Check for WebRTC Transport Requestor endpoint
+        bool isWebRtcEndpoint =
+            (requestPath.endpoint == kWebRTCRequesterDynamicEndpointId && requestPath.cluster == WebRTCTransportRequestor::Id);
+
+        // Only allow these specific endpoints
+        if (!isWebRtcEndpoint)
+        {
+            return CHIP_ERROR_ACCESS_DENIED;
+        }
+
+        if (requestPrivilege != Privilege::kOperate)
+        {
+            return CHIP_ERROR_ACCESS_DENIED;
+        }
+
+        if (subjectDescriptor.authMode != AuthMode::kCase && subjectDescriptor.authMode != AuthMode::kPase &&
```

Comment:
Do we have any reason to validate the authentication mode here?
This is essentialy a "is communication valid" in other places, however I am unclear here of why we would do this. At most we could say `restrict to case` ... or not bother.